Ical.Net/Collections/PeriodCollectionBase.cs

```diff
@@ -0,0 +1,155 @@
+﻿//
+// Copyright ical.net project maintainers and contributors.
+// Licensed under the MIT license.
+//
+
+#nullable enable
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Ical.Net.DataTypes;
+
+namespace Ical.Net.Collections;
+
+/// <summary>
+/// Represents a base class for collections of periods.
+/// </summary>
+public abstract class PeriodCollectionBase : ICollection<Period>
```

Comment:
If we have a `PeriodCollectionBase` shouldn't it be the base class also of `PeriodList`?

Thinking from a different side - the class basically is an implementation of ICollection but doesn't do any additional validation of the entries. Would it make sense to replace the whole class, just like `ExceptionDateCollection` with generic `List<IDateTime>` or the like and implement the specific functionality in extension methods?

Reply:
Good point. I'll try to make `PeriodList` internal as it's only a vehicle for de/serializing. Will have a look.